core/bcast/bcast.go

```diff
@@ -110,6 +110,14 @@ func (b Broadcaster) Broadcast(ctx context.Context, duty core.Duty, set core.Sig
 		return err
 
 	case core.DutyBuilderRegistration:
+		slot, err := firstSlotInCurrentEpoch(ctx, b.eth2Cl)
+		if err != nil {
+			return errors.Wrap(err, "calculate first slot in epoch")
```

Comment:
If this function call fails we'll not submit a builder registration.

What if the upstream beacon node briefly wasn't able to fulfill this request, but later it'll be able to submit a validator registration?

Bottom line is: should we allow validator registration to fail if the resulting metric *might* be off?

Reply:
@gsora yeah good thought, but here's the catch:
We need these three value to calculate the first slot in epoch: `genesis_time`, `slots_per_epoch` and `slot_duration`. `go-eth2-client` caches (since these are constants) these values when we create an instance of `eth2http` and it returns cached values every time we call the associated methods. So I think we should be fine calling these methods here.

Refer:
- https://github.com/attestantio/go-eth2-client/blob/122f335c2f6a20558846872dc4b47cf61ad615d6/http/service.go#L69
- https://github.com/attestantio/go-eth2-client/blob/122f335c2f6a20558846872dc4b47cf61ad615d6/http/service.go#L145

Other approach could be store these constants in `Broadcaster` struct and populate those values in constructor.

Reply:
I think this won't return an error almost all of the time since as you mentioned, these are cached constants, ofc unless there's an error with eth2http configuration itself. So, i think it's safe to catch and return an error.